### PR TITLE
Help Center: Check user eligibility for support

### PR DESCRIPTION
**Pull request by renancarvalho**
Related to https://github.com/Automattic/dotcom-forge/issues/9881
Fixes https://github.com/Automattic/dotcom-forge/issues/9881

## Proposed Changes

* Add check whether the user should talk to a human or get support in forums

## Why are these changes being made?

* Only eligible users can talk to a human, this PR addresses it.

## Testing Instructions

* Use live link and add the flag `?flags=help-center-experience`
* Create a free account and open the Helpcenter
* Start a conversation with Wapuu and ask to "talk to a human"
* You should see this
image

When clicking on the "Ask in our forums" button, you should be redirected to
image

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: packages/odie-client/src/components/message/get-support.tsx [hunk 1]
```diff
@@ -1,39 +1,61 @@
...
 
-		await newConversation();
-	};
-
-	const getButtonText = () => {
-		return shouldUseHelpCenterExperience
-			? __( 'Get instant support', __i18n_text_domain__ )
-			: __( 'Get support', __i18n_text_domain__ );
-	};
+export const GetSupport: React.FC< GetSupportProps > = ( { onClickAdditionalEvent } ) => {
```
Comment (escapemanuele) [inline packages/odie-client/src/components/message/get-support.tsx:16]: I don't think we needed the extra mile here just for one prop 👍

File: packages/odie-client/src/components/message/get-support.tsx [hunk 2]
```diff
@@ -1,39 +1,61 @@
...
+import { useNavigate } from 'react-router-dom';
 import { useOdieAssistantContext } from '../../context';
 import { useCreateZendeskConversation } from '../../hooks';
-
 import './get-support.scss';
 
-export const GetSupport = ( {
-	onClickAdditionalEvent,
-}: {
+interface GetSupportProps {
```
Comment (escapemanuele) [inline packages/odie-client/src/components/message/get-support.tsx:7]: As in the other comment

**Review by escapemanuele — APPROVED** [review]
LGTM. I added a minor comment that is really nit-picky, up to you 👍

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17008741

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @renancarvalho for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17008741).

**Pull request merged**